resources/assets/js/designer/components/svgcanvas.vue

```diff
@@ -37,39 +40,46 @@
             EventBus.$on(actions.designer.flow.creating().type, (value) => this.creatingFlow(value))
             EventBus.$on(actions.designer.flow.create().type, (value) => this.createFlow(value))
             EventBus.$on(actions.designer.shape.remove().type, (value) => this.removeElement(value))
-            EventBus.$on(actions.bpmn.toXML().type, (value) => this.toXML(value))
+            EventBus.$on(actions.bpmn.save().type, (value) => this.saveBPMN(value))
         },
         methods: {
             /**
              * Create the element
              * @param event
              */
             createElement(event) {
-                let name = event.target.id.split(':')
+                let definition = event.target.getAttribute("event")
+                let type = event.target.getAttribute("type")
                 this.updateCoordinates()
                 const defaultOptions = {
-                    id: name[1] + '_' + Math.floor((Math.random() * 100) + 1),
-                    type: name[1],
+                    id: type + '_' + Math.floor((Math.random() * 1000000) + 1),
+                    type: type,
                     bounds: {
                         x: event.x - this.diagramCoordinates.x,
                         y: event.y - this.diagramCoordinates.y
-                    }
+                    },
+                    eventDefinition: definition
                 }
-                if (Elements[name[1].toLowerCase()]) {
+                if (Elements[type.toLowerCase()]) {
                     this.builder.createShape(defaultOptions, true)
                 } else {
-                    console.error(name[1].toLowerCase() + "is not support")
+                    console.error(type.toLowerCase() + " is not support")
```

Comment:
Please use:

ProcessMaker.alert(type.toLowerCase() + ' is not support', 'danger');